spatial-tracking-explainer.md

```diff
@@ -209,35 +209,31 @@ An `XRRay` object includes both an `origin` and `direction`, both given as `DOMP
+Developers can specify application-specific transforms by calling the `getOffsetReferenceSpace()` method of any `XRReferenceSpace`. This returns a new `XRReferenceSpace` with an origin equal to the origin of the parent reference space offset by the by the `position` and `orientation` described by the `XRRigidTransform` passed to `getOffsetReferenceSpace()`.
 
 A common use case for this attribute would be for a "teleportation" mechanic, where the user "jumps" to a new point in the virtual scene, after which the selected point is treated as the new virtual origin which all tracked motion is relative to.
 
 ```js
 // Teleport the user a certain number of meters along the X, Y, and Z axes
 function teleport(deltaX, deltaY, deltaZ) {
-  let currentOrigin = xrReferenceSpace.originOffset;
-  xrReferenceSpace.originOffset = new XRRigidTransform(
-      { x: currentOrigin.position.x + deltaX,
-        y: currentOrigin.position.y + deltaY,
-        z: currentOrigin.position.z + deltaZ },
-      currentOrigin.orientation);
+  xrReferenceSpace = xrReferenceSpace.getOffsetReferenceSpace(
+      new XRRigidTransform({ x: deltaX, y: deltaY, z: deltaZ }));
 }
 ```
 
 ### Relating between reference spaces
 There are several circumstances in which developers may choose to relate content in different reference spaces.
 
 #### Inline to Immersive
-It is expected that developers will often choose to preview `immersive` experiences with a similar experience `inline`. In this situation, users often expect to see the scene from the same perspective when they make the transition from `inline` to `immersive`. To accomplish this, developers should grab the `transform` of the last `XRViewerPose` retrieved using the `inline` session's `XRReferenceSpace` and set it as the `originOffset` of the `immersive` session's `XRReferenceSpace`. The same logic applies in the reverse when exiting `immersive`.
+It is expected that developers will often choose to preview `immersive` experiences with a similar experience `inline`. In this situation, users often expect to see the scene from the same perspective when they make the transition from `inline` to `immersive`. To accomplish this, developers should grab the `transform` of the last `XRViewerPose` retrieved using the `inline` session's `XRReferenceSpace` and pass it to `getOffsetReferenceSpace()` on the `immersive` session's `XRReferenceSpace` to produce an appropriately offset reference space. The same logic applies in the reverse when exiting `immersive`.
```

Comment:
> To accomplish this, developers should grab the `transform` of the last `XRViewerPose` retrieved using the `inline` session's `XRReferenceSpace` and pass it to `getOffsetReferenceSpace()` on the `immersive` session's `XRReferenceSpace` to produce an appropriately offset reference space. The same logic applies in the reverse when exiting `immersive`.

Oh ick.  This means that developers will need to create a throw-away reference space so they can call `getOffsetReferenceSpace()` on it?  Should we be considering allowing an optional `XRRigidTransform` as a parameter to `requestReferenceSpace()`?

Reply:
I doubt that the "throwaway" reference space + one offset one will be all that expensive relative to creating the reference space with the right offset to begin with, but it's a valid API ergonomics issue. It should definitely be handled in a separate issue from this one, however, and I'd argue that it sounds to me like the type of convenience feature that would work really well for a "vNext" edition of the spec.

Reply:
Sounds good.  Perhaps let's file the issue and put it in the "Future" milestone.